### ISSUE TITLE
ioctl() need not return 0 on success

### ISSUE THREAD

**Issue by elahav** [lxqt/qtermwidget#354]
##### Expected Behavior

In kptydevice.c, if the call to ioctl() succeeds then, depending on the system, it may return any value other than -1. Only -1 can be used to infer a failure.

##### Current Behavior

The code assumes that only 0 is a successful return code.

##### Possible Solution

```
@@ -90,7 +90,7 @@ bool KPtyDevicePrivate::_k_canRead()
 #else
     int available;
 #endif
-    if (!::ioctl(q->masterFd(), PTY_BYTES_AVAILABLE, (char *) &available)) {
+    if (::ioctl(q->masterFd(), PTY_BYTES_AVAILABLE, (char *) &available) != -1) {
 #ifdef Q_OS_SOLARIS
         // A Pty is a STREAMS module, and those can be activated
         // with 0 bytes available. This happens either when ^C is
```

##### Steps to Reproduce (for bugs)

1. 
2. 
3. 
4. 

##### System Information

* Distribution & Version: QNX 7.1
* Kernel: 
* Qt Version: 5.6
* lxqt-build-tools Version: N/A
* Package version: N/A

**Comment by yan12125** [comment]
The fix looks good. Mind to create a pull request for that?

**Comment by yan12125** [comment]
Fixed with #357
